lib/fog/vsphere/models/compute/snapshots.rb

```diff
@@ -0,0 +1,31 @@
+require 'fog/core/collection'
+require 'fog/vsphere/models/compute/snapshot'
+
+module Fog
+  module Compute
+    class Vsphere
+      class Snapshots < Fog::Collection
+        attribute :server_id, alias: :instance_uuid
+        attribute :parent_snapshot
+        model Fog::Compute::Vsphere::Snapshot
+
+        def all(filters = {})
+          if parent_snapshot
+            load service.list_child_snapshots(parent_snapshot, filters)
+          else
+            requires :server_id
+            load service.list_vm_snapshots(server_id, filters)
+          end
+        end
+
+        def get(snapshost_ref)
+          all.each do |s|
+            nested = s.get(snapshost_ref)
+            return nested if nested
+          end
+          nil
+        end
```

Comment:
Isn't there an api to get only one snapshot?
You need to load all of them in memory and then sort it out?